graphql/resolvers.go

```diff
@@ -712,6 +712,14 @@ func (r *queryResolver) PatchBuildVariants(ctx context.Context, patchID string)
 	return result, nil
 }
 
+func (r *queryResolver) CommitQueue(ctx context.Context, id string) (*restModel.APICommitQueue, error) {
+	commitQueue, err := r.sc.FindCommitQueueByID(id)
+	if err != nil || commitQueue == nil {
```

Comment:
So reading through the implementation of FindCommitQueueByID() it seems like it's impossible to for err to be not-nil and commitQueue to be nil, to make this less ambiguous, I'd delete the (or).

Reply:
I don't think this is the right answer here, but the 404 case would be uniquely caught by `errors.Cause(err) == err`

Reply:
Sorry maybe i'm misunderstanding but `errors.Cause(err)` resolves to `no commit queue found for '<Some id that was passed in>'`  which is also the same as what `err` resolves to.

So wouldn't, `errors.Cause(err) ==err` always be true.  so in order to catch the 404 case i would have to compare `errors.Cause(err) == "no commit queue found for '<Some id that was passed in> '"`

Reply:
Right, it only resolves to that when the problem is the 404 case, and not when it's because there's a DB connection issue. 

(the only reason it could possibly matter, and I don't know enough about GQL to know if it matters) but there's no real point in retrying 404 errors unless you expect a thing to exist in the future, but most other errors are transient. (As it happens Annie fixed an bug today where not properly categorizing errors lead to bad retry logic.)